### DISCUSSION TITLE
Extract Multiple Rows -> JSON

### DISCUSSION THREAD

**Discussion by ttelik** [Budibase/budibase#4672] [Help] [answered]
Hi Everyone,

I'm looking to extract several rows from a database and send them to an API (as a json formatted array). I am doing this in two different places with two different databases and it seems to work differently in each.

The page structure is as follows:
DataProvider -> Button (Action) -> API Call -> Binding set to {{ DataProvider.Rows }}

In the first case, the data provider returns one row and it works great. The handlebars notation above returns a JSON formatted array with one element containing all the columns for that row.

In the second case, the data provider returns multiple rows and it seems I just have these objects that are impossible to extract data from. I am not using Relationships in either database.

I'm trying to avoid using a repeater and manually pulling out each of the columns, since there are a number of columns.

Am I missing something?

**Comment by mjashanks** [comment, marked as answer]
Hey @ttelik - I'm not sure exactly what you mean.

However, `DataProvider.Rows` should always be an array of items, regardless of how many rows have been returned.

If you want to manipulate this array, or extract certain fields you really need to use Javascript to do so.

Happy to help more, if you tell me 
1. what `{{ DataProvider.Rows }}` is outputting in the second instance  
2. What you want to extract and send to the api

**Reply by ttelik** [reply]
Hi Michael,

So, I've added a paragraph field within the data provider just to illustrate the point.

![image](https://user-images.githubusercontent.com/74260856/155607930-bccc1562-6829-430d-b39c-192104e8a7c9.png)

And the result is:
![image](https://user-images.githubusercontent.com/74260856/155607890-0dc355df-12d7-449d-bb27-9964d51ffdae.png)

I can retrieve .Rows.0 for example, but it just says 'Object'. I would like to extract a json formatted row, which is what I'm getting in another similar data provider on the same page. The only difference between them is a different db and the fact that the one that works only has one row returned.

**Reply by mjashanks** [reply]
Ok got it.... try `{{ JSONstringify RunDataNumeric Data Provider.Rows }}`

Again, if you want to manipulate that in any way, Javascript is your friend!

**Reply by ttelik** [reply]
That seems to throw a handlebars syntax error. I guess what I'm stuck on is the difference between the one db where it works fine, and this one, which seems to return garbage. 

What's the difference?

**Reply by mjashanks** [reply]
I edited my comment btw - got it wrong :)

I don't honestly know the difference - it's just the way handlebars is displaying it

**Reply by ttelik** [reply]
I think something else is wrong here. Here's the output:

![image](https://user-images.githubusercontent.com/74260856/155609714-53e7e5ea-ee64-4d58-9e2a-974c87f1f661.png)

This is the same db where I posted that the row query fails in the automation. discussion #4643

**Reply by ttelik** [reply]
Take that back - it was my parens that broke it. Works now. Thank you!
